controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthInterceptor.java

```diff
@@ -21,25 +21,36 @@
 import io.pravega.auth.AuthConstants;
 import io.pravega.auth.AuthException;
 import io.pravega.auth.AuthHandler;
-import io.pravega.shared.security.token.JsonWebToken;
 import java.security.Principal;
-import java.util.HashMap;
-import java.util.Map;
 
+import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
-import static io.pravega.auth.AuthHandler.Permissions.READ_UPDATE;
-
+/**
+ * Intercepts gRPC requests and sets up Auth context.
+ */
 @Slf4j
-public class PravegaInterceptor implements ServerInterceptor {
-    private static final String AUTH_CONTEXT = "PravegaContext";
-    private static final String INTERCEPTOR_CONTEXT = "InterceptorContext";
-    private static final Context.Key<Principal> AUTH_CONTEXT_TOKEN = Context.key(AUTH_CONTEXT);
-    public static final Context.Key<PravegaInterceptor> INTERCEPTOR_OBJECT = Context.key(INTERCEPTOR_CONTEXT);
+public class AuthInterceptor implements ServerInterceptor {
+
+    private static final String PRINCIPAL_KEY_NAME = "PravegaContext";
+    private static final String INTERCEPTOR_KEY_NAME = "InterceptorContext";
+
+    /**
+     * Represents the key used for indexing the {@link java.security.Principal} object stored in the current
+     * {@link io.grpc.Context}.
+     */
+    static final Context.Key<Principal> PRINCIPAL_OBJECT_KEY = Context.key(PRINCIPAL_KEY_NAME);
```

Comment:
Ok i think i understand this now.
The comment says this is the `Key` for indexing the principal. 
So the key is static and constant. But from the context we can get the value against this key. 
So if i understand this correctly, we store principal and interceptor into the context against some known keys and then we can get the values from `AuthContext`.`key`.get().

Reply:
Yes, exactly! The gRPC Context API is a bit awkward in this respect. 

(I've explained the same here: https://github.com/pravega/pravega/pull/4037#discussion_r308529649)